tests/tests.rs

```diff
@@ -6,9 +6,9 @@ mod loaders;
 
 #[test]
 fn test_era_era() {
-  let net = parse_core("@main = * & * ~ *");
+  let net = parse_core("@main = (* *) & * ~ *");
```

Comment:
hmm, but we have one example that the net doesn't work, right? which is when the root is simply `*`. we ommited this case, but we shouldn't delete tests because the implementation is correct, in fact that's why tests exist!

I suggest going back to the original code(or adding a new test that contains the old case) and either try add `#[ignore]` or figure out the bug in the GPU(if you take more than, say, 30 minutes, add a task for it and also add `#[ignore]` on the test, so that we know something is wrong)